### ISSUE TITLE
Error trying to create a remote database with React Native, Expo

### ISSUE THREAD

**Issue by RonaldBecerra** [pouchdb/pouchdb#8268]
### Issue
I am building an app with Expo, React Native, and I use the library "pouchdb-react-native" to store data. It has been successful with local databases, but now I am trying to create a remote database. According to the [documentation](https://pouchdb.com/guides/databases.html), I could use the instruction:

        var db = new PouchDB('http://localhost:5984/kittens');

The documentation says that the database will not be created until an API call is made. For example, we could ask for its info with:

        db.info();

The problem is that it throws the error:

         Object {
             "message": undefined,
             "name": "unknown",
             "status": 0,
        }

If instead of the word 'localhost' I put the IP of the computer with which I am building the application, the error changes to the text below. (Both the computer and the mobile device are connected to Internet via the same router). I have tried with the IPv4 (that I got with the console using the command "ipconfig"); also with the local IPv6, and the public IPv4 that I got from this [link](https://www.whatismyip.com/what-is-my-public-ip-address/). 

        [Error: ETIMEDOUT]

Also, since the documentation talks about the library 'pouchdb' and not 'pouchdb-react-native', I tried using it, but it throws:

        [TypeError: Network request failed]

I thought the problem could be the port, so I tried enabling it, but the error persists. 

It is important to add that I first had to install CouchDB in my computer. Then I went to the link http://localhost:5984/_utils/ as the [documentation](https://pouchdb.com/guides/setup-couchdb.html) suggested. I first tried with latest version 3.1.1 but there was a problem in that link because all tabs lead to the same LogIn page, and I could not login with the username and password that I had created when installing CouchDB. So I tried with lower version 2.3.1 and it worked fine in that link. I can create databases in there (in the page). But I still cannot do it from the app that is in the mobile device.

I have also asked for this in [Stack Overflow](https://stackoverflow.com/questions/65801682/cannot-connect-to-a-remote-database-pouchdb-in-react-native), and someone suggested that I had to specify a first username and password when creating the database, like:

        var db = new PouchDB('http://myuser:mypassword@localhost:5984/kittens');

but it didn't change anything.

### Info
- Environment: Node.js, React Native, Expo 38.0.0. 
- Platform: Windows 10 in the computer developing the App, Android in the mobile device used for tests.

### Reproduce
I created a provisional React Component class, and I put the code that creates the database in the method "ComponentDidMount":

```
import React, { Component } from 'react';
import PouchDB from 'pouchdb-react-native'
...

export default class Gallery extends Component{
    ...
    componentDidMount(){
        var db1 = new PouchDB('http://localhost:5984/kittens');
        db1.info()
        .then(function (info) {
            console.log("The db1 info is", info);
        }).catch(error => {
            console.log("The db1 error is", error);
        })

        const pouchdb = require('pouchdb');
        var db2 = new pouchdb('http://localhost:5984/kittens');
        db2.info()
        .then(function (info) {
            console.log("The db2 info is", info);
        }).catch(error => {
            console.log("The db2 error is", error);
        })
    }
}
```

**Comment by dheimoz** [comment]
hello @RonaldBecerra are you connecting to a Couchdb 3.x instance? If that is the case, you might need to log in. Admin party is disabled by default. Should you use a ver 2, with admin party enabled, that would be a different story.

**Comment by RonaldBecerra** [comment]
Hello @dheimoz. 

Yes, I am using Couchdb 3.1.1. If I access this direction http://localhost:5984/_utils, it asks me to log in, but I simply cannot. I put the username and password that I created when installing Couchdb and it does not work (and I have uninstalled and reinstalled it several times). With version 2.3.1, that direction does not ask for log in. I can manually create databases in there, but the problem with the mobile app persists. The app cannot create remote databases with the code I put above. I don't know if the code is which is not appropiate.

**Comment by dheimoz** [comment]
Hi @RonaldBecerra 

First things first. You should be aware that starting ver 3.X you need to provide a user to access Couchdb
https://devclass.com/2020/02/27/couchdb-3-0-ends-admin-party-era/

Then, should to need to reset you admin password , to access Faxuton ( http://localhost:5984/_utils ) follow this steps:

1) Locate your etc folder. That depends entirely on your OS and whether you chose default installation or custom. In my case I use Windows 10 and decided to install on C:\CouchDB. The etc folder will be located there.
2) Open the file "local.ini" with your text editor. You might need higher permissions to write on it. In muy case I use right click and  open as administrator. Optionally you can create a backup of this file just in case.
3)  Essentially this file contains all your configuration. Comments use ";" semicolon.  Near the end find  
[admins]
4) There are detailed instructions about resetting the password and even create admin users. Basically you need to change in plain text the new password after the "=" symbol. 
5) Restart CouchDB Service. The password will be encrypted.
6) Check you can access  Faxuton with the new credentials. http://localhost:5984/_utils 

After you have confirmed access you need to be aware that CouchDB has 4 ways to authenticate and access the Databases. 

A) Basic Authentication, you can utilize user and password in the url "http://user:pass@localhost:5984/dbname
You can also Attach autorization headers with btoa(user:pass)
https://docs.couchdb.org/en/stable/api/server/authn.html#basic-authentication

B) Cookie Authentication, for this you need to post a request to the authentication end point
https://docs.couchdb.org/en/stable/api/server/authn.html#cookie-authentication

C) Proxy Authentication: Personally I haven't use it.
https://docs.couchdb.org/en/stable/api/server/authn.html#proxy-authentication

D) JWT Authentication: This is new and needs to configure with the previous ini file.
https://docs.couchdb.org/en/stable/api/server/authn.html#jwt-authentication

I would start with Basic Authentication maybe with PostMan or curl. That way you can avoid CORS issues.

Hope this helps.

**Comment by dheimoz** [comment]
Sorry, I misread some of your comments. You want to access Couchdb form a different device, right? Then, after you have confirmed you have access to Faxuton, make sure there is no firewall blocking the connection to that port (3984). Also enable CORS using Faxuton.

I am terribly sorry, but I must say that this seems not to be an actual issue with Pouchdb, but something related to your environment.

Hope you can confirm. Remember to close your issue if it is fixed.

**Comment by RonaldBecerra** [comment]
@dheimoz Thanks for your comments. I finally managed to login to CouchDB via Fauxton, and the only way to do it was editing the _local.ini_ file that you mentioned. I also enabled CORS using the instruction: "add-cors-to-couchdb http://localhost:5984 -u username -p password" (which was only possible once I had logged in to Couchdb). But I still cannot access to it from the mobile device with React Native and I have already restarted the computer.

The port 5984 should be enabled because I added an entry rule in the Firewall. I tried enabling it with both protocols TCP and UDP. I don't know if I should close the issue because it is probably no longer related to PouchDB, but the reality is that the problem has not been solved yet.
